member-service/src/main/java/com/hedvig/memberservice/services/customerio/CustomerIOTrustpilotEventListener.kt

```diff
@@ -0,0 +1,97 @@
+package com.hedvig.memberservice.services.customerio
+
+import com.hedvig.integration.notificationService.NotificationService
+import com.hedvig.memberservice.events.EmailUpdatedEvent
+import com.hedvig.memberservice.events.NameUpdatedEvent
+import com.hedvig.memberservice.query.MemberEntity
+import com.hedvig.memberservice.query.MemberRepository
+import com.hedvig.memberservice.services.trustpilot.TrustpilotReviewService
+import com.hedvig.resolver.LocaleResolver
+import com.neovisionaries.i18n.CountryCode
+import org.axonframework.config.ProcessingGroup
+import org.axonframework.eventhandling.EventHandler
+import org.springframework.context.annotation.Profile
+import org.springframework.stereotype.Component
+import java.util.Locale
+import java.util.Objects
+
+@Component
+@Profile("customer.io")
+@ProcessingGroup("CustomerIOTrustpilotBackfill")
+class CustomerIOTrustpilotEventListener(
+    private val notificationService: NotificationService,
+    private val memberRepository: MemberRepository,
+    private val trustpilotReviewService: TrustpilotReviewService
+) {
+
+    @EventHandler
+    fun on(evt: NameUpdatedEvent) {
+        val member = memberRepository.findById(evt.memberId).orElseThrow {
+            IllegalStateException("No member found for ${evt.memberId}")
+        }
```

Comment:
A bit more clear logging IMO
```suggestion
        val member = memberRepository.findById(evt.memberId).orElseThrow {
            IllegalStateException("No member found when generating Trustpilot link (memberId=${event.memberId})")
        }
```

Reply:
Right - I figured the stack trace would be enough but even clearer messages are always good